src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -329,16 +331,18 @@ public AgentSettings LoadSettings()
             bool runAsService = command.GetRunAsService();
             if (runAsService)
             {
-                if (!new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator))
-                {
-                    Trace.Error("Needs Administrator privileges for configure agent as windows service.");
-                    throw new SecurityException(StringUtil.Loc("NeedAdminForConfigAgentWinService"));
-                }
-
                 Trace.Info("Configuring to run the agent as service");
                 var serviceControlManager = HostContext.GetService<IWindowsServiceControlManager>();
                 serviceControlManager.ConfigureService(agentSettings, command);
             }
+            //This will be enabled with AutoLogon code changes are tested
+            else
+            {                
+                ConfigureAutoLogonIfNeeded(command);
```

Comment:
That decision needs to be revisited. It doesn't make sense to have diverging patterns.